### PR TITLE
[JENKINS-60562] Remove unused, deprecated CLI classes.

### PR DESCRIPTION
**Pull request by jeffret-b**
See [JENKINS-60562](https://issues.jenkins-ci.org/browse/JENKINS-60562).

These two classes have been deprecated and are no longer used since the Remoting-based CLI mechanism was removed almost a year ago. That had been deprecated for a while before that. I'd like to see if we can remove these also.

In a search of the full jenkinsci organization I only found 1 reference to these files. [jenkins.py](https://github.com/jenkinsci/jenkins.py/blob/cbfc2ec231b447638d2d3e3bf82e3bc7bb160549/python-wrapper/src/main/java/jenkins/python/expoint/CliTransportAuthenticatorPW.java) has a reference to `CliTransportAuthenticatorPW` but there's not much that can be done with it. jenkins.py hasn't been updated in a long time.

### Proposed changelog entries

* Remove unused, deprecated CLI classes (issue [JENKINS-60562](https://issues.jenkins-ci.org/browse/JENKINS-60562))

### Submitter checklist

- [n/a] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
No tests needed for removal of unused code.
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

**Review by jvz — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
jenkins.py is used by just one plugin, and personally I do not see a big deal in breaking it taking the current situation with the wrapper plugin after JEP-200. Ideally it would be great to have a formal deprecation process for it via JEP, but it is outside of the scope for this ticket

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Requesting changes to indicate that I wait for a response from @jeffret-b before the merge

**Comment by jeffret-b** [comment]
What are you asking for, Oleg? A JEP connected to jenkins.py? A formal deprecation process?

(It would be great for us to establish a formal deprecation process. We really get stuck with deprecated stuff in many ways.)

File: core/src/main/java/hudson/cli/CliTransportAuthenticator.java
Comment (jglick) [inline core/src/main/java/hudson/cli/CliTransportAuthenticator.java]: Fine, this is completely unused. (Formerly used only by a module, but we have removed that reference and released a new version of the module and bundled it.)

File: core/src/main/java/hudson/cli/Connection.java
Comment (jglick) [inline core/src/main/java/hudson/cli/Connection.java]: This is in active use by the CloudBees proprietary plugin `operations-center-agent` so please do not delete it without some compatibility plan in that plugin. (I think it would suffice to just copy its implementation there.)

**Review by jglick — CHANGES_REQUESTED** [review]
`jenkins.py` is a red herring, that just mirrors every type in core for whatever weird reason. Ignore it.

File: core/src/main/java/hudson/cli/Connection.java
Comment (jeffret-b) [reply core/src/main/java/hudson/cli/Connection.java]: Maybe I should just exclude `Connection` from this PR and continue with `CliTransportAuthenticator`. WDYT?

Comment (jglick) [reply core/src/main/java/hudson/cli/Connection.java]: Fine, or just coördinate with @jtnord etc.

File: core/src/main/java/hudson/cli/CliTransportAuthenticator.java
Comment (jglick) [reply core/src/main/java/hudson/cli/CliTransportAuthenticator.java]: I think I declined to remove this in #3838 due to timing issues, FYI.

Comment (jeffret-b) [reply core/src/main/java/hudson/cli/CliTransportAuthenticator.java]: Timing issues of code execution? Or of removing it at the same time as the others? 

My intent here was to see if we could accept the removal now that more time has elapsed. Maybe we should just wait on both of these.

Comment (jglick) [reply core/src/main/java/hudson/cli/CliTransportAuthenticator.java]: > of removing it at the same time as the others

This, because I think there was still an outstanding reference at the time. Now it should be fine to remove.

**Pull request closed without merging**
